code/__HELPERS/game.dm

```diff
@@ -213,34 +199,19 @@
 
 	return
 
-/// Returns a list of hearers in view(view_radius) from source (ignoring luminosity). recursively checks contents for hearers
+/// Returns a list of hearers in view(view_radius) from source (ignoring luminosity). uses important_recursive_contents[RECURSIVE_CONTENTS_HEARING_SENSITIVE]
 /proc/get_hearers_in_view(view_radius, atom/source)
-
 	var/turf/center_turf = get_turf(source)
 	. = list()
 	if(!center_turf)
 		return
-	var/list/processing_list = list()
-	if (view_radius == 0) // if the range is zero, we know exactly where to look for, we can skip view
-		processing_list += center_turf.contents // We can shave off one iteration by assuming turfs cannot hear
-	else
-		var/lum = center_turf.luminosity
-		center_turf.luminosity = 6 // This is the maximum luminosity
-		var/target = source.loc == center_turf ? source : center_turf //this is reasonably faster if true, and very slightly slower if false
-		for(var/atom/movable/movable in view(view_radius, target))
-			if(movable.flags_1 & HEAR_1) //dont add the movables returned by view() to processing_list to reduce recursive iterations, just check them
-				. += movable
-				SEND_SIGNAL(movable, COMSIG_ATOM_HEARER_IN_VIEW, processing_list, .)
-			processing_list += movable.contents
-		center_turf.luminosity = lum
-
-	var/i = 0
-	while(i < length(processing_list)) // recursive_hear_check inlined here, the large majority of the work is in this part for big contents trees
-		var/atom/atom_to_check = processing_list[++i]
-		if(atom_to_check.flags_1 & HEAR_1)
-			. += atom_to_check
-			SEND_SIGNAL(atom_to_check, COMSIG_ATOM_HEARER_IN_VIEW, processing_list, .)
-		processing_list += atom_to_check.contents
+	var/lum = center_turf.luminosity
+	center_turf.luminosity = 6 // This is the maximum luminosity
+	for(var/atom/movable/movable in view(view_radius, center_turf))
+		if(movable.important_recursive_contents && movable.important_recursive_contents[RECURSIVE_CONTENTS_HEARING_SENSITIVE]) //dont add the movables returned by view() to processing_list to reduce recursive iterations, just check them
+			. += movable.important_recursive_contents[RECURSIVE_CONTENTS_HEARING_SENSITIVE]
+			SEND_SIGNAL(movable, COMSIG_ATOM_HEARER_IN_VIEW, .)
```

Comment:
Out of scope, but this signal is not the best way to implement this.
Not asking you to change it because you're already doing enough here, but this should be instead handled when the message is sent, with this movable receiver retransmitting it.

Reply:
i agree and only dullahans use it (everything their head hears they hear), getting rid of it would be nice but thats a target for later